app/src/main/java/org/xbmc/kore/ui/sections/remote/NowPlayingFragment.java

```diff
@@ -752,6 +753,21 @@ private void setNowPlayingInfo(PlayerType.GetActivePlayersReturnType getActivePl
         }
 
         mediaTitle.setText(title);
+        mediaTitle.post(new Runnable() {
+            @Override
+            public void run() {
+                int lines = mediaTitle.getLineCount();
+                int maxLines = TextViewCompat.getMaxLines(mediaTitle);
+                if (lines > maxLines) {
+                    mediaTitle.setHorizontallyScrolling(true);
```

Comment:
This causes the textview to be a single line for text that needs more than max lines even when not scrolling. How about setting it based on the selected state of the `TextView`?

```
                if (lines > maxLines) {
                    mediaTitle.setClickable(true);
                    mediaTitle.setOnClickListener(new View.OnClickListener() {
                        @Override
                        public void onClick(View v) {
                            if (v.isSelected()) {
                                ((TextView) v).setMaxLines(1);
                                mediaTitle.setHorizontallyScrolling(true);
                            } else {
                                ((TextView) v).setMaxLines(2);
                                mediaTitle.setHorizontallyScrolling(false);
                            }
                            v.setSelected(!v.isSelected()); }
                    });
                }
            }
        });
```

Reply:
This causes the layout (posters, buttons, descriptions, etc) to reposition as the `TextView` switches between 1 and 2 lines height. Is that a concern?

Otherwise am happy to update. I found I needed a few more changes to get better control of the ellipsis (the ellipsis is presumably hidden when `lines > maxLines` and the mode is marquee by default as it would be generated on the hidden last line):

```java
// default ellipsize style <item name="android:ellipsize">end</item>
if (lines == maxLines && mediaTitle.getLayout().getEllipsisCount(lines - 1) > 0) {
    mediaTitle.setClickable(true);
    mediaTitle.setOnClickListener(new View.OnClickListener() {
        @Override
        public void onClick(View v) {
            v.setSelected(!v.isSelected());
            TextUtils.TruncateAt ellipsize;
            if (v.isSelected()) {
                ellipsize = TextUtils.TruncateAt.MARQUEE;
            } else {
                ellipsize = TextUtils.TruncateAt.END;
            }
            mediaTitle.setEllipsize(ellipsize);
            mediaTitle.setHorizontallyScrolling(v.isSelected());
        }
    });
}
```
```

Reply:
I don't mind that it jumps now. It's only when the user clicks the title to make it scroll.
And nice touch changing the ellipsize to END to make the user aware there is more text. 
There is a small problem here that the END ellipsize is only shown when the user switches back from marquee to multiline. If I remember correctly you probably have to add it programmatically as setting the ellipsize to END in the style will brake the marquee functionality.

Reply:
Have pushed an update, hopefully I haven't misunderstood the desired effect.

Reply:
Works great! @SyncedSynapse any objections?

Regarding the refactoring. It would be nice to put the runnable in a separate method. Preferably in the [UIUtils](https://github.com/xbmc/Kore/blob/4dda3ac6e85b84dc2205950eea9b4269edf75c97/app/src/main/java/org/xbmc/kore/utils/UIUtils.java) class. That would also reduce the code duplication, as I believe both runnables are the same.